src/backend/preprocessor/Preprocessor.js

```diff
@@ -0,0 +1,113 @@
+define([
+    './cfgGenerator',
+    './envGenerator'
+], function(cfgGenerator, envGenerator) {
+
+    var _ = require('underscore');
+
+    function AstToCfg(_astObj) {
+	this.astObj = _astObj;
+	this.preprocessed = {};
+    }
+    
+    AstToCfg.prototype.retrieveGlobals = function retrieveGlobals() {
+	// retrieve globals
+	this.preprocessed.global = {};
+	for(var i = 0; i < this.astObj.external_declarations.length; i++) {
+	    if(this.astObj.external_declarations[i].type !== "declaration")
+		continue;
+	    var varDef = this.astObj.external_declarations[i];
+	    // duplicate code, extract it
+	    var tvalue = varDef.tvalue;
+	    var varEntry;
+	    if(tvalue.type === 'concrete_type') {
+		varEntry = {
+		    type: tvalue.name
+		};
+	    } else if(tvalue.type === 'pointer') {
+		varEntry = {
+		    type: 'pointer',
+		    of: tvalue.tvalue.name // make it recursive
+		};
+	    }
+	    // end of duplicate code
+	    this.preprocessed.global[varDef.name] = varEntry;
+	}
+    };
+
+    AstToCfg.prototype.collectPrototypes = function collectPrototypes() {
+	this.preprocessed.prototypes = {};
+
+	for(var i = 0; i < this.astObj.external_declarations.length; i++) {
+	    if(this.astObj.external_declarations[i].type !== 'function_definition')
+		continue;
+
+	    var funcDef = this.astObj.external_declarations[i];
+	    this.preprocessed.prototypes[funcDef.declaration.name] = funcDef.declaration;
+	}
+    };
+
+    AstToCfg.prototype.generateFunctions = function generateFunctions() {
+	// generate functions
+	this.preprocessed.functions = {};
+	this.preprocessed.values = {};
+	for(var i = 0; i < this.astObj.external_declarations.length; i++) {
+	    if(this.astObj.external_declarations[i].type !== "function_definition")
+		continue;
+	    var funcDef = this.astObj.external_declarations[i];
+	    var funcDecl = funcDef.declaration;
+	    
+	    var args = [];
+	    for(var j = 0; j < funcDecl.param_names.length; j++)
+		args.push(funcDecl.name + '_PARAMETER_' + funcDecl.param_names[j]);
+
+	    var envAndValues = envGenerator(funcDef);
+	    var env = envAndValues.env;
+	    var values = {};
+	    for(var val in envAndValues.constants) {
+		values[envAndValues.constants[val]] = isNaN(val) ? val : Number.parseInt(val); // replace it with smarter check maybe?
```

Comment:
TypeError: Object function Number() { [native code] } has no method 'parseInt'

there is a function `parseInt` in global namespace, never heard about `Number.parseInt` though. I always just use the `Number` constructor, as in: `Number('12')`